api/client/src/main/java/org/projectnessie/client/auth/oauth2/AuthorizationCodeFlow.java

```diff
@@ -149,15 +172,19 @@ public Tokens fetchNewTokens() {
   private void doRequest(HttpExchange exchange) {
     LOGGER.debug("Authorization Code Flow: received request");
     inflightRequestsPhaser.register();
-    requestFuture.complete(exchange);
+    redirectUriFuture.complete(exchange);
     tokensFuture
         .handle((tokens, error) -> doResponse(exchange, error))
         .whenComplete((v, error) -> exchange.close())
         .whenComplete((v, error) -> inflightRequestsPhaser.arriveAndDeregister());
   }
 
   private Void doResponse(HttpExchange exchange, Throwable error) {
-    LOGGER.debug("Authorization Code Flow: sending response");
+    if (LOGGER.isDebugEnabled()) {
+      LOGGER.debug(
+          "Authorization Code Flow: sending response, error: {}",
+          error == null ? "none" : error.toString());
```

Comment:
nit: why not just pass `error` to the logger? It should handle nulls.. Is it so much worse to have "null" in a DEBUG message instead of "none"?

Reply:
Same, IntelliJ won't like it if it is used as a message argument. If it's not a message argument, then the whole stack trace will be printed, which is overkill.

Reply:
noted, nevermind